packages/expo-file-system/ios/EXFileSystem/EXSessionTasks/EXSessionResumableDownloadTaskDelegate.m

```diff
@@ -0,0 +1,60 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import <EXFileSystem/EXSessionResumableDownloadTaskDelegate.h>
+
+@interface EXSessionResumableDownloadTaskDelegate ()
+
+@property (strong, nonatomic, readonly) EXDownloadDelegateOnWriteCallback onWriteCallback;
+@property (weak, nonatomic) EXResumableManager *manager;
+@property (strong, nonatomic) NSString *uuid;
+
+@end
+
+@implementation EXSessionResumableDownloadTaskDelegate
+
+- (instancetype)initWithResolve:(UMPromiseResolveBlock)resolve
+                         reject:(UMPromiseRejectBlock)reject
+                       localUrl:(NSURL *)localUrl
+             shouldCalculateMd5:(BOOL)shouldCalculateMd5
+                onWriteCallback:(EXDownloadDelegateOnWriteCallback)onWriteCallback
+                resumableManager:(EXResumableManager *)manager
+                           uuid:(NSString *)uuid;
+{
+  if (self = [super initWithResolve:resolve
+                             reject:reject
+                           localUrl:localUrl
+                 shouldCalculateMd5:shouldCalculateMd5]) {
+    _onWriteCallback = onWriteCallback;
+    _manager = manager;
+    _uuid = uuid;
+  }
+  return self;
+}
+
+- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error
+{
+  if (error) {
+    // The task was paused by us. So, we shouldn't throw.
+    if (error.code == NSURLErrorCancelled) {
+      self.resolve([NSNull null]);
+    } else {
+      self.reject(@"ERR_FILE_SYSTEM_UNABLE_TO_DOWNLOAD",
+                  [NSString stringWithFormat:@"Unable to download file: %@", error.description],
+                  error);
+    }
+    
+    [_manager unregisterTask:_uuid];
```

Comment:
Wouldn't we want to unregister the task in case there was no error too? (Shouldn't we move it to outside of the `if (error)` block? 🤔